perun-core/src/main/java/cz/metacentrum/perun/core/impl/UsersManagerImpl.java

```diff
@@ -67,6 +72,32 @@
 		"user_ext_sources.modified_by as user_ext_sources_modified_by, user_ext_sources.modified_at as user_ext_sources_modified_at, " +
 		"user_ext_sources.created_by_uid as ues_created_by_uid, user_ext_sources.modified_by_uid as ues_modified_by_uid";
 
+	private static Map<String, Pattern> userExtSourcePersistentPatterns;
+
+	static {
+		// Prepare userExtSourcePersistentPatterns for matching regex from perun property file.
+		// It is done in advance because of performance.
+		try {
+			userExtSourcePersistentPatterns = new HashMap<>();
+			String persistentConfig = BeansUtils.getPropertyFromConfiguration("perun.userExtSources.persistent");
+			for (String extSource : persistentConfig.split(";")) {
+				String[] extSourceTuple = extSource.split(",", 2);
+				if (extSourceTuple.length > 1) {
+					userExtSourcePersistentPatterns.put(extSourceTuple[0], Pattern.compile(extSourceTuple[1]));
+				} else {
+					userExtSourcePersistentPatterns.put(extSource, Pattern.compile(".*"));
+				}
+			}
+		} catch (InternalErrorException e) {
+			log.info("Error when reading property perun.userExtSources.persistent. Serving default behavior.", e);
+			// If error occurred no persistent user ext sources are considered. e.g. property is not set.
+			userExtSourcePersistentPatterns = new HashMap<>();
+		} catch (PatternSyntaxException e) {
+			// cannot use checked exception because of static context
+			throw new InternalErrorRuntimeException("Error when parsing perun.userExtSources.persistent property. Probably regex is in wrong format.", e);
```

Comment:
The PatternSyntaxException can raise, only if "split" argument (which should be regex) is wrongly formatted. In other words, it's not dependent on input (i.e. property from configuration).  
I can guarantee you, that ";" is correct regex :-), therefor this exception never raises.  
Please either delete this "catch" section or at least change the error message in InternalErrorRuntimeException.